plugins/out_s3/s3.c

```diff
@@ -2112,6 +2127,11 @@ static void cb_s3_flush(struct flb_event_chunk *event_chunk,
     struct s3_file *upload_file = NULL;
     struct flb_s3 *ctx = out_context;
     struct multipart_upload *m_upload_file = NULL;
+    msgpack_unpacked result;
+    msgpack_object *obj;
+    size_t off = 0;
+    struct flb_time tms;
+    time_t file_first_log_time = NULL;
```

Comment:
Does this compile?  Like I noted in my comment above, you are using a time_t pointer. And here you set it to NULL. So pointer is needed.

Reply:
this comment still applies